src/caffe/net.cpp

```diff
@@ -79,10 +79,17 @@ void Net<Dtype>::Init(const NetParameter& in_param) {
     }
     // Setup layer.
     const LayerParameter& layer_param = param.layer(layer_id);
+    if (layer_param.propagate_down_size() > 0) {
+      CHECK_EQ(layer_param.propagate_down_size(),
+          layer_param.bottom_size())
+      << "propagate_down param must be specified"
```

Comment:
this and next line should have 4-space indents, and next line string should start with a space (`" either`)